k9mail-library/src/test/java/com/fsck/k9/mail/store/imap/ImapResponseHelper.java

```diff
@@ -3,16 +3,37 @@
 
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 
 import com.fsck.k9.mail.filter.PeekableInputStream;
 
 
 public class ImapResponseHelper {
+
+    @SafeVarargs
+    public static List<List<ImapResponse>> createMultipleImapResponses(List<String>... responses) throws IOException {
```

Comment:
The selected state commands split the commands and execute them individually, so you get a `List<List<ImapResponse>>`. These are then parsed by the response classes and combined into a single response object which is returned.